3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js

```diff
@@ -15,9 +18,10 @@ const rollDice = require('../../helpers/pokerDiceRoller');
 function rollTheDices() {
   // TODO Refactor this function
   const dices = [1, 2, 3, 4, 5];
-  return rollDice(1);
+  const myAllPromises = dices.map((element) => rollDice(element));
+  Promise.all(myAllPromises)
+    
 }
-
 rollTheDices()
   .then((results) => console.log('Resolved!', results))
   .catch((error) => console.log('Rejected!', error.message));
```

Comment:
This code doesn't work. Have you actually tried it? Here is what I get:

```
10:26:19.762 Dice 1 scheduled for 5 rolls...
10:26:19.763 Dice 1 is now: ACE
10:26:19.763 Dice 2 scheduled for 8 rolls...
10:26:19.763 Dice 2 is now: TEN
10:26:19.763 Dice 3 scheduled for 7 rolls...
10:26:19.764 Dice 3 is now: ACE
10:26:19.764 Dice 4 scheduled for 8 rolls...
10:26:19.764 Dice 4 is now: QUEEN
10:26:19.764 Dice 5 scheduled for 4 rolls...
10:26:19.764 Dice 5 is now: ACE
Something went wrong: Cannot read property 'then' of undefined
10:26:20.259 Dice 1 is now: JACK
10:26:20.260 Dice 4 is now: ACE
10:26:20.266 Dice 3 is now: TEN
10:26:20.267 Dice 2 is now: ACE
10:26:20.268 Dice 5 is now: JACK
10:26:20.755 Dice 1 is now: NINE
10:26:20.757 Dice 4 is now: JACK
10:26:20.768 Dice 3 is now: NINE
10:26:20.770 Dice 2 is now: KING
10:26:20.772 Dice 5 is now: NINE
10:26:21.251 Dice 1 is now: QUEEN
10:26:21.253 Dice 4 is now: NINE
10:26:21.270 Dice 3 is now: KING
10:26:21.274 Dice 2 is now: NINE
10:26:21.276 Dice 5 is now: QUEEN
10:26:21.276 Dice 5 settles on QUEEN in 4 rolls.
10:26:21.747 Dice 1 is now: ACE
10:26:21.747 Dice 1 settles on ACE in 5 rolls.
10:26:21.750 Dice 4 is now: QUEEN
10:26:21.772 Dice 3 is now: ACE
10:26:21.778 Dice 2 is now: TEN
10:26:22.247 Dice 4 is now: ACE
10:26:22.275 Dice 3 is now: TEN
10:26:22.281 Dice 2 is now: ACE
10:26:22.744 Dice 4 is now: JACK
10:26:22.744 Dice 4 continues rolling on the floor...
(node:28142) UnhandledPromiseRejectionWarning: Error: Dice 4 rolled off the table.
    at rollOnce (/home/jim/dev/hackyourfuture/homework-fork/3-UsingAPIs/helpers/pokerDiceRoller.js:76:18)
    at Timeout._onTimeout (/home/jim/dev/hackyourfuture/homework-fork/3-UsingAPIs/helpers/pokerDiceRoller.js:89:28)
    at listOnTimeout (internal/timers.js:554:17)
    at processTimers (internal/timers.js:497:7)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:28142) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:28142) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
10:26:22.778 Dice 3 is now: NINE
10:26:22.778 Dice 3 continues rolling on the floor...
10:26:22.778 Dice 3 settles on NINE in 7 rolls.
10:26:22.784 Dice 2 is now: KING
10:26:22.784 Dice 2 continues rolling on the floor...
10:26:23.240 Dice 4 is now: NINE
10:26:23.240 Dice 4 settles on NINE in 8 rolls.
10:26:23.287 Dice 2 is now: NINE
```

Reply:
I had a small change about it, i forget to check  it ,now it works regularly.